astropy/table/tests/test_df.py

```diff
@@ -0,0 +1,656 @@
+
+                if backend == "polars" and "f2" in column:
+                    # No Polars Float16 support
+                    pass
+                else:
+                    assert t2_dtype == t_dtype
+
+                # Pandas-specific checks
+                if backend == "pandas":
+                    # Pandas-specific exact comparison
+                    assert_array_equal(d[column], t[column])
+                    if t[column].dtype.isnative:
+                        assert d[column].dtype == t[column].dtype
+                    else:
+                        assert d[column].dtype == t[column].dtype.newbyteorder()
+
+        # Pandas-specific endian tests (skip if not pandas)
+        if not backend == "pandas":
+            return
+
+        # Regression test for astropy/astropy#1156 - the following code gave a
+        # ValueError: Big-endian buffer not supported on little-endian
+        # compiler. We now automatically swap the endian-ness to native order
+        # upon adding the arrays to the data frame.
+        # Explicitly testing little/big/native endian separately -
+        # regression for a case in astropy/astropy#11286 not caught by #3729.
+        d[["<i4", ">i4"]]
+        d[["<f4", ">f4"]]
+
+        # Additional round-trip checks
+        for column in t.columns:
+            if column in ("u", "s"):
+                assert_array_equal(t2[column], t[column])
+            else:
+                assert_allclose(t2[column], t[column])
+            if t[column].dtype.isnative:
+                assert t2[column].dtype == t[column].dtype
+            else:
+                assert t2[column].dtype == t[column].dtype.newbyteorder()
+
+    def test_from_df_simple(self, backend, use_legacy_pandas_api):
+        data = {"a": [1, 2, 3], "b": [4.0, 5.0, 6.0], "c": ["x", "y", "z"]}
+        match backend:
+            case "pandas":
+                import pandas as pd
+
+                df = pd.DataFrame(data)
+            case "polars":
+                import polars as pl
+
+                df = pl.DataFrame(data)
+            case "pyarrow":
+                import pyarrow as pa
+
+                df = pa.Table.from_pydict(data)
+            case "dask":
+                import dask.array as da
+                import dask.dataframe as dd
+
+                df = dd.concat(
+                    [
+                        dd.from_dask_array(
+                            da.from_array(arr, chunks="auto"), columns=col
+                        )
+                        for col, arr in data.items()
+                    ],
+                    axis=1,
+                )
+            case "duckdb":
+                import duckdb
+
+                df = duckdb.read_json(
+                    io.StringIO(
+                        json.dumps(
+                            [
+                                dict(zip(data.keys(), values))
+                                for values in zip(*data.values())
```

Comment:
while correct, this pattern seems a bit fragile, let's make it harder to break silently
```suggestion
                                dict(zip(data.keys(), values, strict=True))
                                for values in zip(*data.values(), strict=True)
```